MessagingLibrary/RabbitMq/RabbitMqPublisher.cs

```diff
@@ -0,0 +1,44 @@
+﻿using MessagingLibrary.Interfaces;
+using RabbitMQ.Client;
+using System.Text;
+using System.Text.Json;
+
+namespace MessagingLibrary.RabbitMq;
+
+public class RabbitMqPublisher : IMessagePublisher, IDisposable
+{
+    private readonly IConnection _connection;
+    private readonly IModel _channel;
+    private readonly string _exchange;
+
+    public RabbitMqPublisher(string hostname, string exchange)
+    {
+        _exchange = exchange;
+
+        var factory = new ConnectionFactory { HostName = hostname };
+        _connection = factory.CreateConnection();
+        _channel = _connection.CreateModel();
```

Comment:
for thread safety, it would make more sense to create a connection and  a channel per publish, these should be moved to the publish method instead.

Reply:
also then you can do : 
using var connection
using var channel